### PR TITLE
Add Graal Native Image description in JDK compatibility section.

### PR DESCRIPTION
**Pull request by chaitanyawaikar**
I have added the description of GraalVM in the JDK compatibility section. Please let me know if you would like me to make any changes. Looking forward for your feedback. Thanks

### CODE REVIEW

**Comment by SethTisue** [comment]
@andreaTP @olofwalker LGTY?

File: _overviews/jdk-compatibility/overview.md [hunk 1]
```diff
@@ -48,7 +50,7 @@ In almost every case, you're free to use the JDK and JVM of your choice.
 
 JDK 8 users typically use the Oracle JDK or some flavor of OpenJDK.
 
-Most JDK 11+ users are using either OpenJDK or GraalVM.
+Most JDK 11+ users are using either OpenJDK or [GraalVM](https://www.graalvm.org/docs/reference-manual/scala/) which runs in the context of OpenJDK. Developers can use [sbt-native-image](https://github.com/scalameta/sbt-native-image) which uses GraalVM under the hood.
```
Comment (olofwalker) [inline _overviews/jdk-compatibility/overview.md:53]: This needs clarification. Are you referring to the GraalVM compiler which using JVMCI can be utilized as a C2 compiler by OpenJDK?

Comment (chaitanyawaikar) [reply _overviews/jdk-compatibility/overview.md:53]: To be very frank, I have not gone through the source code of `sbt-native-image`. I just read that it uses GraalVm under the hood. If this line is confusing, I can completely remove it. Else, if you would like to have an alternate explanation, I would be happy to put it.
Thanks

Comment (andreaTP) [reply _overviews/jdk-compatibility/overview.md:53]: `sbt-native-image` mention deserves a separate section ( and you can add sbt-native-packager https://www.scala-sbt.org/sbt-native-packager/formats/graalvm-native-image.html )

We can mention that there are several known projects already compiling with the GraalVM AOT compiler and therefore running on SubstrateVM.

@chaitanyawaikar this is always the confusion with Graal:

 - AOT compiler
 - JIT compiler (as opposed to C2 in a normal JVM)
 - Polyglot runtime
 
 I think your addition covers pretty well the JIT part!

Comment (chaitanyawaikar) [reply _overviews/jdk-compatibility/overview.md:53]: @andreaTP does your comment mean that my changes look good ? Sorry, but I am still confused :(

Comment (SethTisue) [reply _overviews/jdk-compatibility/overview.md:53]: ping @andreaTP

Comment (andreaTP) [reply _overviews/jdk-compatibility/overview.md:53]: sorry, missed the comment, not really for this PR I suggest that @chaitanyawaikar simply remove the line:

> Developers can use [sbt-native-image](https://github.com/scalameta/sbt-native-image) which uses GraalVM under the hood.

and I will follow up with the changes for `native-image`, works for you @chaitanyawaikar ?

Comment (chaitanyawaikar) [reply _overviews/jdk-compatibility/overview.md]: Thanks for the clarification @andreaTP . Removed the line

**Review by andreaTP — APPROVED** [review]

**Comment by SethTisue** [comment]
thank you!

**Pull request merged**